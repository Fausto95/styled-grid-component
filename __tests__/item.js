import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Item from '../src/item';

test('it works order - 2', () => {
  const tree = renderer.create(<Item order="2" />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('order', '2');
});

test('it works order - 3', () => {
  const tree = renderer.create(<Item order="3" />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('order', '3');
});

test('it works basis', () => {
  const tree = renderer.create(<Item basis="30%" />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('flex-basis', '30%');
});

test('it works grow', () => {
  const tree = renderer.create(<Item grow />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('flex-grow', '1');
});

test('it works shrink', () => {
  const tree = renderer.create(<Item shrink />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('flex-shrink', '1');
});

test('it works noShrink', () => {
  const tree = renderer.create(<Item noShrink />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('flex-shrink', '0');
});
