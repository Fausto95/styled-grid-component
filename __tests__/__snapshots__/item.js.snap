// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works basis 1`] = `
.c0 {
  -webkit-flex-basis: 30%;
  -ms-flex-preferred-size: 30%;
  flex-basis: 30%;
}

<div
  className="c0"
/>
`;

exports[`it works grow 1`] = `
.c0 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<div
  className="c0"
/>
`;

exports[`it works noShrink 1`] = `
.c0 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

<div
  className="c0"
/>
`;

exports[`it works order - 2 1`] = `
.c0 {
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
}

<div
  className="c0"
  order="2"
/>
`;

exports[`it works order - 3 1`] = `
.c0 {
  -webkit-order: 3;
  -ms-flex-order: 3;
  order: 3;
}

<div
  className="c0"
  order="3"
/>
`;

exports[`it works shrink 1`] = `
.c0 {
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
}

<div
  className="c0"
/>
`;
